Validation/RecoTau/src/TauValidationMiniAOD.cc

```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
+        pTOverProng_dm6Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 10) {
+        mtau_dm10Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm10Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 11) {
+        mtau_dm11Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm11Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      }
+
+      // fill decay mode population plot
+      ntau_vs_dmMap.find("")->second->Fill(taus->size(), matchedTau->decayMode());
+
+      //Fill decay mode migration 2D histogragms
+      //First do a gen Matching
+      unsigned genindex = 0;
+      for (std::vector<reco::GenParticle>::const_iterator genParticle = genParticles->begin();
+           genParticle != genParticles->end();
+           genParticle++) {
+        if (abs(genParticle->pdgId()) == 15) {
+          float gendR = deltaR(matchedTau->eta(), matchedTau->phi(), genParticle->eta(), genParticle->phi());
+          if (gendR < gendRmin) {
+            gendRmin = gendR;
+            genmatchedTauIndex = genindex;
+          }
+        }
+        genindex = genindex + 1;
+      }
+
+      if (gendRmin < 0.15) {
+        int nPhotonsFromTauDecay = 0;
+        int nPi0s = 0;
+        int nPis = 0;
+        auto &gentau = genParticles->at(genmatchedTauIndex);
+        for (unsigned idtr = 0; idtr < gentau.numberOfDaughters(); idtr++) {
+          const reco::GenParticle *dtr = dynamic_cast<const reco::GenParticle *>(gentau.daughter(idtr));
+          int dtrpdgID = std::abs(dtr->pdgId());
+          int dtrstatus = dtr->status();
+          if (dtrpdgID == 12 || dtrpdgID == 14 || dtrpdgID == 16)
+            continue;
+          if (dtrpdgID == 111 || dtrpdgID == 311)
+            nPi0s++;
+          else if (dtrpdgID == 211 || dtrpdgID == 321)
+            nPis++;
+          else if (dtrpdgID == 22) {
+            float dr_taugamma = deltaR(gentau.eta(), gentau.phi(), dtr->eta(), dtr->phi());
+            if (dr_taugamma < 0.3 && dtr->pt() > 2)
+              nPhotonsFromTauDecay++;
+          } else if (dtrpdgID == 15 && dtrstatus == 2 /*&& dtr->isLastCopy()*/) {
+            for (unsigned idtr2 = 0; idtr2 < dtr->numberOfDaughters(); idtr2++) {
+              const reco::GenParticle *dtr2 = dynamic_cast<const reco::GenParticle *>(dtr->daughter(idtr2));
+              int dtr2pdgID = std::abs(dtr2->pdgId());
+              if (dtr2pdgID == 12 || dtr2pdgID == 14 || dtr2pdgID == 16)
+                continue;
+              if (dtr2pdgID == 111 || dtr2pdgID == 311)
+                nPi0s++;
+              else if (dtr2pdgID == 211 || dtr2pdgID == 321)
+                nPis++;
+              else if (dtr2pdgID == 22) {
+                float dr_taugamma = deltaR(gentau.eta(), gentau.phi(), dtr2->eta(), dtr2->phi());
```

Comment:
Using `deltaR2` is computationally cheaper (here and elsewhere in this code)

Reply:
Done